### ISSUE TITLE
How to provide a provides DB for a PkgBase repo

### ISSUE THREAD

**Issue by igalic** [rosorio/pkg-provides#7]
Hi!

I'm running https://alpha.pkgbase.live/
an unofficial repository for the [FreeBSD PkgBase](https://wiki.freebsd.org/PkgBase) project, which packages the FreeBSD base as individual packages.

How can i provide a pkg-provides database for these packages, too?

**Comment by rosorio** [comment]
Hi,

I can suggest a couple of little things we can do to make it in a faster and
unclean way, but it's probably time to start integrating pkg-provides with existing FreeBSD tools such as poudriere for the database generation.

I was procrastinating on this task for a while, but this can be a good opportunity to start working on this task.

If this match your I can try to give you a feedback around June and see how it goes.

Regards
-- rodrigo

**Comment by igalic** [comment]
thank you for the response!
I am in no hurry.

i have meanwhile learned from @bapt that the provides db is based on [`locate(1)`](https://man.freebsd.org/locate(1)).
When you try to better integrate it with existing tools, are you gonna migrate it to SQLite?

**Comment by rosorio** [comment]
Yes, the database uses the  bigram algorithm, which is the one used by locate.
So I take advantage of the  base system tools to generate the database, and specially the
/usr/libexec/locate.mklocatedb script. In the  plugin, I integrate the bigram code itself.

In my first implementation I try to use sqlite, but it was to slow to insert the 1304465
records and rebuilding the index required if you plan to do full text searches.
In fact bigram is quite fast and reduce the size if the database without too much effort.

database dump  (text only)    : 691M
xz database dump                   :   32M
bigram db                                 :   96M
xz bigram db                            :   17M

Of course, I'm open to discussions if you think another way to improve the situation :)

**Comment by rosorio** [comment]
Hi,
I push two simple scripts to generate the pkg-provides database after a poudriere build.
I think you can use them to generate your pkgbase database.

https://github.com/rosorio/pkg-provides/tree/master/poudriere

I put some details about how to test it with pkg-provides plugin.
